File: paivita_henkilo.py
import psycopg2

def paivita_henkilo(tietokantayhteys, HenkiloID, Etunimi, Sukunimi):
    try:
        cursor = tietokantayhteys.cursor()
        # Päivitetään henkilön etunimi ja sukunimi ID:n perusteella
        sql = "UPDATE Henkilot SET Etunimi = %s, Sukunimi = %s WHERE HenkiloID = %s;"
        cursor.execute(sql, (Etunimi, Sukunimi, HenkiloID))
        
        # Tarkistetaan, onko rivi päivitetty
        if cursor.rowcount > 0:
            print(f"Henkilön tiedot päivitetty ID: {HenkiloID}. Etunimi: {Etunimi}, Sukunimi: {Sukunimi}")
        else:
            print(f"Henkilöä ID: {HenkiloID} ei löytynyt, ei päivitetty.")
        
        tietokantayhteys.commit()
    except (Exception, psycopg2.Error) as error:
        print("Virhe henkilön päivityksessä:", error)

def main():
    try:
        # Yhdistetään tietokantaan
        yhteys = psycopg2.connect(
            dbname="postgres",
            user="postgres",
            password="postgres",
            host="localhost",
            port="5432"
        )

        # Kysytään käyttäjältä henkilön ID, etunimi ja sukunimi
        HenkiloID = input("Anna päivitettävän henkilön ID: ")
        Etunimi = input("Anna henkilön uusi etunimi: ")
        Sukunimi = input("Anna henkilön uusi sukunimi: ")

        # Päivitetään henkilön tiedot
        paivita_henkilo(yhteys, HenkiloID, Etunimi, Sukunimi)

        # Suljetaan tietokantayhteys
        yhteys.close()
    except (Exception, psycopg2.Error) as error:
        print("Virhe tietokantayhteydessä:", error)

if __name__ == "__main__":
    main()

File: paivita_henkilo.php
<?php
// Tietokantayhteyden asetukset
$host = "localhost"; // PostgreSQL-palvelimen osoite
$port = "5432";      // PostgreSQL-portti (oletus)
$dbname = "postgres"; // Tietokannan nimi
$user = "postgres";  // Käyttäjänimi
$password = "postgres";      // Salasana

// Luo yhteys PostgreSQL:ään
$conn = pg_connect("host=$host port=$port dbname=$dbname user=$user password=$password");

// Tarkista yhteys
if (!$conn) {
    die("Yhteyden muodostaminen epäonnistui: " . pg_last_error());
}

// Tarkista onko id ja muut kentät lähetetty lomakkeella
if ($_SERVER["REQUEST_METHOD"] == "POST") {
    $henkiloid = $_POST['henkiloid'];  // Oletetaan, että id lähetetään lomakkeella
    $etunimi = $_POST['etunimi'];      // Etunimi
    $sukunimi = $_POST['sukunimi'];    // Sukunimi

    // Päivitä henkilot-taulu
    $sql = "UPDATE henkilot SET etunimi = $1, sukunimi = $2 WHERE henkiloid = $3";
    $result = pg_query_params($conn, $sql, array($etunimi, $sukunimi, $henkiloid));

    if ($result) {
        echo "Tietoja on päivitetty onnistuneesti.";
    } else {
        echo "Virhe tietojen päivityksessä: " . pg_last_error($conn);
    }
}

pg_close($conn); // Sulje yhteys
?>

<!-- HTML-lomake -->
<!DOCTYPE html>
<html lang="fi">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Päivitä Henkilö</title>
</head>
<body>
    <h1>Päivitä Henkilön Tiedot</h1>
    <form action="paivita_henkilo.php" method="post">
        <label for="henkiloid">Henkilön ID:</label>
        <input type="number" id="henkiloid" name="henkiloid" required><br><br>
        
        <label for="etunimi">Etunimi:</label>
        <input type="text" id="etunimi" name="etunimi" required><br><br>

        <label for="sukunimi">Sukunimi:</label>
        <input type="text" id="sukunimi" name="sukunimi" required><br><br>

        <input type="submit" value="Päivitä">
    </form>
</body>
<div class="links">
        <a href="index.php">Etusivu</a>
    </div>
</html>
